src/Smd.php

```diff
@@ -293,10 +303,10 @@ public function isDojoCompatible()
      *
      * @param Smd\Service|array $service
      * @return self
-     * @throws Exception\RuntimeException
-     * @throws Exception\InvalidArgumentException
+     * @throws RuntimeException
+     * @throws InvalidArgumentException
      */
-    public function addService($service)
+    public function addService($service) : self
```

Comment:
We need to consider here just allowing `Smd\Service` here and then we can drop checks inside. Thoughts?

Reply:
I agree, this would make the purpose of `$service` more precise. I will open an issue.